tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11236,1074 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+    if constexpr (std::is_same_v<T, float>)
+      return FloatDist(generator);
+    if constexpr (std::is_same_v<T, double>)
+      return DoubleDist(generator);
+    if constexpr (std::is_same_v<T, uint16_t>)
+      return Uint16Dist(generator);
+    if constexpr (std::is_same_v<T, uint32_t>)
+      return Uint32Dist(generator);
+    if constexpr (std::is_same_v<T, uint64_t>)
+      return Uint64Dist(generator);
+
+    VERIFY_FAIL("Unsupported data type for generate()");
+  }
+
+private:
+  // Mersenne Twister 'random' number generator. Generated numbers are based on
+  // the seed value and are deterministic for any given seed.
+  std::mt19937 generator;
+  std::uniform_int_distribution<int16_t> Int16Dist;
+  std::uniform_int_distribution<int32_t> Int32Dist;
+  std::uniform_int_distribution<int64_t> Int64Dist;
+  std::uniform_int_distribution<DirectX::PackedVector::HALF> DXHalfDist;
+  std::uniform_int_distribution<uint16_t> Uint16Dist;
+  std::uniform_int_distribution<uint32_t> Uint32Dist;
+  std::uniform_int_distribution<uint64_t> Uint64Dist;
+  std::uniform_real_distribution<float> FloatDist;
+  std::uniform_real_distribution<double> DoubleDist;
+
+  //  The ranges for generation. A std::uniform_real_distribution can only have
+  //  a range that is equal to the types largest value. This is due to precision
+  //  issues. So instead we define some large values.
+  const float FLOAT_RANGE_MIN = -1e20f;
+  const float FLOAT_RANGE_MAX = 1e20f;
+  const double DOUBLE_RANGE_MIN = -1e100;
+  const double DOUBLE_RANGE_MAX = 1e100;
+};
+
+template <typename T>
+bool DoVectorsMatch(const std::vector<T> &VecInput,
+                    const std::vector<T> &VecExpected, float Tolerance) {
+  // Sanity check. Ensure both vectors have the same size
+  if (VecInput.size() != VecExpected.size()) {
+    VERIFY_FAIL(L"Vectors are different sizes!");
+    return false;
+  }
+
+  // Stash mismatched indexes for easy failure logging later
+  std::vector<size_t> MismatchedIndexes;
+  for (size_t Index = 0; Index < VecInput.size(); ++Index) {
+    if constexpr (std::is_same_v<T, HLSLBool_t>) {
+      // Compiler was very picky and wanted an explicit case for any T that
+      // doesn't implement the operators in the below else. ( > and -). It
+      // wouldn't accept putting this constexpr as an or case with other
+      // statements.
+      if (VecInput[Index] != VecExpected[Index]) {
+        MismatchedIndexes.push_back(Index);
+      }
+    } else if (Tolerance == 0 && VecInput[Index] != VecExpected[Index]) {
+      MismatchedIndexes.push_back(Index);
+    } else if constexpr (std::is_same_v<T, HLSLBool_t>) {
+      // Compiler was very picky and wanted an explicit case for any T that
+      // doesn't implement the operators in the below else. ( > and -). It
+      // wouldn't accept putting this constexpr as an or case with other
+      // statements.
+      if (VecInput[Index] != VecExpected[Index]) {
+        MismatchedIndexes.push_back(i);
+      }
```

Comment:
On line 11584, it pushes back `i`, but there is no variable called `i`.  Wondering how this compiles?  Maybe this code is dead?

Reply:
Ahh - so there's two HLSLBool_t arms here, so only the first one is probably being looked at.

Reply:
Yup, that second else if constexpr shouldn't be there anymore. It must be taken out by the compiler because this same code referencing 'i' is still present and building locally for me. 

Fixed.